File: KIWI-Backend/src/main/java/com/d205/KIWI_Backend/menu/domain/MenuCategory.java
package com.d205.KIWI_Backend.menu.domain;

public enum MenuCategory {
    DECAFFEINATED("디카페인"),
    SMOOTHIES_AND_FRAPPES("스무디&프라페"),
    NEW_PRODUCTS("신상품"),
    ADE("에이드"),
    BEVERAGES("음료"),
    COFFEE_COLD_BREW("커피(콜드브루)"),
    COFFEE_HOT("커피(HOT)"),
    COFFEE_ICE("커피(ICE)"),
    TEA("티"),
    DESSERT("디저트");

    private final String displayName;

    MenuCategory(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static boolean isValidCategory(String category) {
        for (MenuCategory menuCategory : MenuCategory.values()) {
            if (menuCategory.getDisplayName().equals(category)) {
                return true;
            }
        }
        return false;
    }
}

File: KIWE-Android/domain/src/main/java/com/kiwe/domain/model/MenuCategory.kt
package com.kiwe.domain.model

import kotlinx.serialization.Serializable

enum class MenuCategory(
    val displayName: String,
    val group: MenuCategoryGroup,
) {
    NEW("신상품", MenuCategoryGroup.NEW),
    COFFEE_ICE("커피(ICE)", MenuCategoryGroup.COFFEE),
    COFFEE_HOT("커피(HOT)", MenuCategoryGroup.COFFEE),
    COLD_BREW("커피(콜드브루)", MenuCategoryGroup.COFFEE),
    DECAF("디카페인", MenuCategoryGroup.DECAF),
    SMOOTHIE_FRAPPE("스무디&프라페", MenuCategoryGroup.SMOOTHIE),
    ADE("에이드", MenuCategoryGroup.DRINKS),
    TEA("티", MenuCategoryGroup.TEA),
    DRINK("음료", MenuCategoryGroup.DRINKS),
    DESSERT("디저트", MenuCategoryGroup.DESSERT),
}

enum class MenuCategoryGroup(
    val displayName: String,
) {
    NEW("신상품"),
    COFFEE("커피"),
    DECAF("디카페인"),
    SMOOTHIE("스무디"),
    TEA("티"),
    DRINKS("음료"),
    DESSERT("디저트"),
}

@Serializable
data class MenuCategoryParam(
    val id: Int,
    val category: String,
    val categoryNumber: Int,
    val hotOrIce: String,
    val name: String,
    val price: Int,
    val description: String,
    val imgPath: String,
)
